frontend/src/devicelist/__tests__/DevicesTable.test.jsx

```diff
@@ -9,40 +9,118 @@ import {
+import StatusIndicator from "../../general/StatusIndicator";
+import DeviceDetailsButton from "../DeviceDetailsButton";
+import * as SampleData from "../../api/SampleData";
 
 Enzyme.configure({ adapter: new Adapter() });
 jest.spyOn(global.console, "error");
 
 describe("<DevicesTable/> component", () => {
   let wrapper;
+  let wrapperInstance;
+  const dummyTitle = "test";
+  const dummyDevices = [];
+
+  beforeEach(() => {
+    wrapper = Enzyme.shallow(
+      <DevicesTable title={dummyTitle} devices={dummyDevices} />
+    );
+    wrapperInstance = wrapper.instance();
+  });
 
   afterEach(() => {
-    jest.clearAllMocks();
+    wrapper.unmount();
   });
 
   describe("has the correct components", () => {
-    const dummyTitle = "TEST TITLE";
-    const dummyDevices = [];
-    beforeEach(() => {
-      wrapper = Enzyme.shallow(
-        <DevicesTable title={dummyTitle} devices={dummyDevices} />
-      );
-    });
-    it("contains one Box component", () => {
+    it("contains 1 <Box/> component", () => {
       expect(wrapper.find(Box)).toHaveLength(1);
     });
-    it("contains one TableContainer component and has maxHeight of 570", () => {
+    it("contains 1 <TableContainer/> component", () => {
       expect(wrapper.find(TableContainer)).toHaveLength(1);
-      expect(wrapper.find(TableContainer).prop("style")).toHaveProperty(
-        "maxHeight",
-        570
+    });
+    it("contains 1 <MaterialTable/> component with expected props", () => {
+      const table = wrapper.find(MaterialTable);
+      expect(table).toHaveLength(1);
+
+      const shallowWrapper = wrapper.instance();
+      const expected = {
+        title: dummyTitle,
+        components: shallowWrapper.components,
+        columns: shallowWrapper.columns,
+        data: dummyDevices,
+        detailPanel: shallowWrapper.detailPanel,
+        options: shallowWrapper.options,
+        icons: shallowWrapper.icons
+      };
+
+      const tableProps = wrapper.find(MaterialTable).props();
+      expect(tableProps).toEqual(expected);
+    });
+  });
+
+  describe("components variable", () => {
+    it(`should have a render() function that returns a <MTableToolbar/> component`, () => {
+      const wrapperProps = wrapper.props();
+      /*  eslint-disable react/jsx-props-no-spreading */
+      const expectedRenderedComponent = (
+        <div className="lightestGrey">
+          <MTableToolbar {...wrapperProps} />
+        </div>
+      );
+
+      expect(wrapperInstance.components.Toolbar(wrapperProps)).toEqual(
+        expectedRenderedComponent
+      );
+    });
+  });
+
+  describe("columns variable", () => {
+    it(`should have a render() function that returns a <StatusIndicator/> component in Status column`, () => {
+      const dummyData = {
+        status: "Online"
+      };
+
+      const expectedRenderedComponent = (
+        <StatusIndicator status={dummyData.status} />
+      );
+
+      expect(wrapperInstance.columns[2].render(dummyData)).toEqual(
```

Comment:
is it better to use `toStrictEqual` here? that's generally what we've been using to check deeper equalities..